api/providers.go

```diff
@@ -113,3 +118,85 @@ func loadInternalProviders(cfg *config.Config, DCOSTools dcos.Tooler) (internalC
 		HTTPEndpoints: httpEndpoints,
 	}, nil
 }
+
+func LoadCollectors(cfg *config.Config, tools dcos.Tooler, client *http.Client) ([]collectors.Collector, error) {
+	providers, err := loadProviders(cfg, tools)
+	if err != nil {
+		return nil, fmt.Errorf("could not init bundle handler: %s", err)
+	}
+	var coll []collectors.Collector
+
+	role, err := tools.GetNodeRole()
+	if err != nil {
+		return nil, fmt.Errorf("could not get role: %s", err)
+	}
+
+	// set filename if not set, some endpoints might be named e.g., after corresponding unit
+	for _, endpoint := range providers.HTTPEndpoints {
+		if !roleMatched(role, endpoint.Role) {
+			continue
+		}
+
+		fileName := fmt.Sprintf("%d-%s.json", endpoint.Port, util.SanitizeString(endpoint.URI))
+		if endpoint.FileName != "" {
+			fileName = endpoint.FileName
+		}
+		//TODO(janisz): Handle socket-only endpoints e.g., dcos-diagnostics
```

Comment:
It's only for DC/OS EE where some services are hidden behind adminrouter and use sockets for communication.

Reply:
But how did we handle that before?

Reply:
We goes thru admin router. We can do it the same right now but there is no point in doing it because we want be independent from adminrouter problems.

Reply:
Understood. So as long as we don't handle socket-only endpoints this is not usable in production, right?

Reply:
It's working on OSS and on EE it needs to go thru adminrouter so configured hostname should be set to external ip.